pkg/sources/github/github.go

```diff
@@ -544,13 +567,27 @@ func (s *Source) cloneRepo(ctx context.Context, repoURL string, installationClie
 		if err != nil {
 			return "", nil, fmt.Errorf("error cloning repo %s: %w", repoURL, err)
 		}
-	default:
-		var token string
-		user, token, err := s.UserAndToken(ctx, installationClient)
+
+	case *sourcespb.GitHub_GithubApp:
```

Comment:
This should be functionally equivalent to what was here before. The only change is in the Github_Token section below, which will skip fetching a user and token if the user and token are both known.

Reply:
Can we do the same thing for `GithubApp` too or does it only make sense for a `Token`?

Reply:
We can eventually. And actually, the Github App token has an `ExpiresAt` field, which will make that decision relatively easy. But for now I'm just doing this for Github Token, because we can't currently refresh that token anyway, so once we have it, we never need to fetch it again.